src/Telegram/Types/ChatMember.php

```diff
@@ -15,131 +22,27 @@
  */
 class ChatMember extends TelegramTypes
 {
+    private const CHILD_CLASS_REFERENCES = [
+        ChatMemberOwner::class,
+        ChatMemberAdministrator::class,
+        ChatMemberMember::class,
+        ChatMemberRestricted::class,
+        ChatMemberLeft::class,
+        ChatMemberBanned::class,
+    ];
```

Comment:
I would like for the actual Telegram classes to remain as clean as possible from 'external' code for two reasons: 
* If Telegram decides to create a field called `created` it won't clash with the defined function.
* It will prevent our users from actually depending on this code, people do crazy stuff and any changes to this will be BC.

I'm thinking that a better option would be to perhaps create a `TelegramAPI\Telegram\Custom\ChatMemberFactory` factory and make the `bindToObject` in `GetChatMember` (and others that return a ChatMemer type) call that dedicated factory instead.

Reply:
![One eternity later](https://media1.tenor.com/m/OrYokHfK4CAAAAAC/movie-one-eternity-later.gif)

Ok, I would need to take a closer look, but I believe the function `mapSubObjects` was created for this purpose.